src/wallet/index.ts

```diff
@@ -386,6 +383,103 @@ class Wallet extends EventEmitter {
         };
       }),
     );
+  }
+    /**
+   * Gets transactions history
+   * @param chainID - chainID to get balances for
+   * @returns history
+   */
+
+  async transactionsLog(chainID: number): Promise<TransactionsLog> {
+    const TXOs = await this.TXOs(chainID);
+    const tmpHistory: TransactionsLog = {};
+
+    // loop through each TXO and add it to the history
+    TXOs.forEach((txOutput) => {
+      const token = formatToByteLength(txOutput.note.token, 32, false);
+      // If we don't have an entry for this token yet, create one
+      if (!tmpHistory[token]) {
+        tmpHistory[token] = []
+      }
+      // first add entry for the token when it was received
+      tmpHistory[token].push({
+        txid: txOutput.txid,
+        amount: txOutput.note.value,
+        direction: TransferDirection.Incoming
+      })
+      // then add another entry if it was spent
+      if(txOutput.spendtxid) {
+        tmpHistory[token].push({
+          txid: txOutput.spendtxid,
+          amount: txOutput.note.value,
+          direction: TransferDirection.Outgoing
+        })
+      }
+    });
+
+    // eslint-disable-next-line func-names
+    const groupBy = function<T extends Record<string, any>, K extends keyof T>(
```

Comment:
Please replace the 'any' with a type here.

We've had a dozen bugs appear because of usage of 'any' keyword, which defeats the purpose of typescript.

Reply:
Changed the groupBy code

Reply:
Fixed